package-res/resources/web/prompting/PromptPanel.js

```diff
@@ -354,6 +354,10 @@ define(['cdf/lib/Base', 'cdf/Logger', 'dojo/number', 'dojo/i18n', 'common-ui/uti
               return paramValue.toUpperCase() != paramSelectedValue.toUpperCase();
             case "java.sql.Date": // Set time to zero to eliminate its influence on the days comparison
               return (new Date(paramValue).setHours(0,0,0,0)) != (new Date(paramSelectedValue).setHours(0,0,0,0));
+            case "[Ljava.lang.String;": // Comparing String arrays
```

Comment:
The reporting engine and server side parameter handling guarantees arrays as types for multi-select parameters.  
All multi-select parameters expect their values as arrays and will not accept creepy modern stuff like Lists, Sets or so on. 

Likewise with contents of the arrays - we get them from the server as strings, and as we only perform an equals comparison, we can treat them as black-box values. As long as they compare correctly we will handle them correctly.